testing/agent-exporter/src/main/java/io/opentelemetry/javaagent/testing/exporter/AgentTestingCustomizer.java

```diff
@@ -30,9 +43,73 @@ public void customize(AutoConfigurationCustomizer autoConfigurationCustomizer) {
 
     autoConfigurationCustomizer.addMeterProviderCustomizer(
         (meterProvider, config) ->
-            meterProvider.registerMetricReader(
-                PeriodicMetricReader.builder(AgentTestingExporterFactory.metricExporter)
-                    .setInterval(Duration.ofMillis(100))
-                    .newMetricReaderFactory()));
+            meterProvider.registerMetricReader(StartableMetricReader.INSTANCE));
+  }
+
+  @Override
+  public void afterAgent(
+      Config config, AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {
+    StartableMetricReader.INSTANCE.start();
+  }
+
+  @SuppressWarnings("ImmutableEnumChecker")
+  private enum StartableMetricReader implements MetricReaderFactory, MetricReader {
```

Comment:
can you add a brief comment explaining why this is needed (or just pointing to the PR discussion)?